modules_core/suggest-mentions.js

```diff
@@ -40,18 +45,36 @@ exports.create = function (api) {
 
   function suggest_search (query) {
     return function (cb) {
-      if(/^[@%]\w/.test(query)) {
-        api.signified(query, function (_, names) {
-          cb(null, names.map(function (e) {
+      if(/^@\w/.test(query)) {
+        api.signified(query, (err, names) => {
+          if(err) return cb(err)
+
+          cb(null, names.map(e => {
+            const { name, rank, id } = e
             return {
-              title: e.name + ':'+e.id.substring(0, 10),
-              value: e.id,
-              subtitle: e.rank,
-              rank: e.rank
+              title: name,
+              subtitle: `(${rank}) ${id.substring(0,10)}`,
+              value: '['+name+']('+id+')',
+              rank,
+              image: api.avatar_image_src(id),
+              showBoth: true
             }
           }))
         })
+      } else if (/^%\w/.test(query)) {
```

Comment:
bit of duplication across these - might make sense to make a bunch of functions in here to give to different modules.